app/src/main/java/hm/orz/chaos114/android/tumekyouen/usecase/InsertDataTask.kt

```diff
@@ -1,49 +1,55 @@
 package hm.orz.chaos114.android.tumekyouen.usecase
 
-import hm.orz.chaos114.android.tumekyouen.network.TumeKyouenService
+import hm.orz.chaos114.android.tumekyouen.db.entities.TumeKyouen
+import hm.orz.chaos114.android.tumekyouen.network.TumeKyouenV2Service
+import hm.orz.chaos114.android.tumekyouen.network.models.Stage
 import hm.orz.chaos114.android.tumekyouen.repository.TumeKyouenRepository
 import io.reactivex.Single
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class InsertDataTask @Inject constructor(
-    private val tumeKyouenService: TumeKyouenService,
+    private val tumeKyouenV2Service: TumeKyouenV2Service,
     private val tumeKyouenRepository: TumeKyouenRepository
 ) {
     var running = false
         private set
 
-    fun run(startStageNo: Int, count: Int): Single<Int> {
+    fun run(currentMaxStageNo: Int, count: Int): Single<Int> {
         if (running) {
             return Single.error(ExclusiveException())
         }
         running = true
         return Single.create { emitter ->
-            var stageNo = startStageNo
-            for (i in 1..count) {
-                val response = tumeKyouenService.getStage(stageNo).blockingGet()
-                val responseString = response.body()?.string()
-                if (responseString == null || "no_data" == responseString) {
-                    emitter.onSuccess(stageNo)
-                    return@create
+            GlobalScope.launch {
+                val response = tumeKyouenV2Service.getStages(currentMaxStageNo + 1, count * 10)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/noboru-i/kyouen-android/blob/d76166622a61443941d2dca457636ccc1842689f/app/src/main/java/hm/orz/chaos114/android/tumekyouen/usecase/InsertDataTask.kt#L28"></span>This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>